cni.go

```diff
@@ -154,16 +154,39 @@ func (c *libcni) Setup(ctx context.Context, id string, path string, opts ...Name
 	return c.createResult(result)
 }
 
+type asynchAttachResult struct {
+	res *types100.Result
+	err error
+}
+
+func asynchAttach(ctx context.Context, n *Network, ns *Namespace, wg *sync.WaitGroup, rc chan asynchAttachResult) {
+	defer wg.Done()
+	r, err := n.Attach(ctx, ns)
+	rc <- asynchAttachResult{res: r, err: err}
+}
+
 func (c *libcni) attachNetworks(ctx context.Context, ns *Namespace) ([]*types100.Result, error) {
+	var wg sync.WaitGroup
```

Comment:
Can we use errgroup?

https://pkg.go.dev/golang.org/x/sync/errgroup

Reply:
I suppose so yes.  I was actually trying to avoid canceling context on first error.. My thought was we might gain some benefit from letting the other in flight network setup/attach requests continue on and populate those successful network attach results.

What do you think?